### DISCUSSION TITLE
dynamic mapped classes not being applied - next.js

### DISCUSSION THREAD

**Discussion by KatuGT** [tailwindlabs/tailwindcss#12483] [Help] [answered]
Hello! here I am trying to create a button with variants and this is what I tried 

```
import { colors } from "./colors";

export const variantesArrowBtn = {
  blanco: {
    textColor: "text-white group-hover:text-verdeOscuro",
    bgArrow: "bg-white",
    colorArrow: colors.verdeOscuro,
  },
  verdeOscuro: {
    textColor: "text-white ",
    bgArrow: "bg-verdeOscuro",
    colorArrow: "#fff",
  },
  rosaBlanco: {
    textColor: "text-white",
    bgArrow: "bg-rosa",
    colorArrow: "#fff",
  },
  rosa: {
    textColor: "text-rosa group-hover:text-white",
    bgArrow: "bg-rosa",
    colorArrow: "#fff",
  },
  bordeado: {
    textColor: "text-white",
    bgArrow: "bg-transparent border border-solid border-white",
    colorArrow: "#fff",
  },
};
```

-------------------------

```
"use client";
import React, { useEffect,  useState } from "react";
import FlechaSVG from "../../public/SVG/Flecha";
import Link from "next/link";
import { variantesArrowBtn } from "@/utils/btnArrowVariants";

const BotonFlecha = ({ texto, variante }) => {
  const [btnVariant, setBtnVariant] = useState({
    textColor: "",
    bgArrow: "",
    colorArrow: "",
  });


  useEffect(() => {
    const style = variantesArrowBtn[variante];
    setBtnVariant(style);
  }, [variante]);

  return (
    <Link
      href={"#"}
      className="group boton-flecha-container flex relative justify-center items-center w-fit h-[40px] px-[34px]"
    >
      <div
        className={`${btnVariant?.bgArrow} absolute h-[40px] w-[40px] rounded-full top-0 left-0 group-hover:w-full transition-all`}
      ></div>
      <div className="h-[40px] w-[40px] z-10 absolute top-0 left-0">
        <FlechaSVG color={btnVariant?.colorArrow} />
      </div>
      <p
        className={`${btnVariant?.textColor}  font-neue font-normal text-[18px] leading-[20px] 
        translate-x-[17px] group-hover:-translate-x-0 z-10 transition-all`}
      >
        {texto}
      </p>
    </Link>
  );
};

export default BotonFlecha;
```

I also tried doing it like ${variantesArrowBtn[variante].bgArrow}, thats the way that is done in the documentation https://tailwindcss.com/docs/content-configuration#dynamic-class-names
I know the classes are working because if I add them just like "text-rosa" "bg-verdeOscuro" bla bla... the work correctly

the **Console** shows no error btw

---------------------------

tailwind.config has some custom colors

```
theme: {
    extend: {
      backgroundImage: {
        "gradient-radial": "radial-gradient(var(--tw-gradient-stops))",
        "gradient-conic":
          "conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))",
      },
      colors: {
        verdeClaro: "#3AC692",
        verdeClaro2: "#61D1A8",
        verdeClaro3: "#89DDBE",
        verdeClaro4: "#B0E8D3",
        verdeClaro5: "#D8F4E9",
        verdeMedio: "#287f66",
        verdeOscuro: "#15383a",
        rosa: "#d95d8b",
        violeta: "#774a62",
        rosaViejo: "#a85477",
        oscuro: "#333237",
        violetaGris: "#46414e",
      },
      fontFamily: {
        neue: ["var(--font-neue)"],
      },
    },
  },
```

**Comment by KatuGT** [comment]
I already restarted the server and even my computer (?  the problem is still here

**Comment by wongjn** [comment, marked as answer]
I assume when you say the "dynamic mapped classes not being applied", you mean that the expected visual effects of the classes are not presenting themselves in the browser, and not that the classes are not existing on the rendered element(s)? If this assumption is correct, consider checking that the file that contains your first code snippet (the code that has `export const variantesArrowBtn`) is covered by the `content` file globs in your Tailwind configuration. Otherwise, consider providing a remotely-hosted git repository that reproduces the unexpected behavior.

**Reply by KatuGT** [reply]
content: [
    "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/components/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/app/**/*.{js,ts,jsx,tsx,mdx}",
    "./public/SVG/**/*.{js,ts,jsx,tsx,mdx}",
    **"./src/utils/**/*.{js,ts,jsx,tsx,mdx}",**
  ],
  
  That was the problem! I just added  **"./src/utils/**/*.{js,ts,jsx,tsx,mdx}",** and now its working!    thank you wongjn!
